extension-manifest-v3/src/background/stats.js

```diff
@@ -230,24 +230,27 @@ chrome.runtime.onMessage.addListener((msg, sender, sendResponse) => {
 
   if (
     __PLATFORM__ === 'safari' &&
-    sender.tab?.id &&
-    sender.frameId !== undefined
+    sender.url &&
+    sender.frameId !== undefined &&
+    sender.tab?.id
   ) {
-    // We cannot trust that Safari fires "chrome.webNavigation.onCommitted"
-    // with the correct tabId (sometimes it is correct, sometimes it is 0).
-    // Thus, let the content_script fire it.
-    if (sender.url && msg.action === 'onCommitted') {
-      setupTabStats(sender.tab.id, parse(sender.url).domain);
-      return false;
-    }
-
-    if (msg.action === 'updateTabStats') {
-      return updateTabStats(
-        sender.tab.id,
-        msg.urls.map((url) =>
-          Request.fromRawDetails({ url, sourceUrl: sender.url }),
-        ),
-      );
+    switch (msg.action) {
+      // We cannot trust that Safari fires "chrome.webNavigation.onCommitted"
+      // with the correct tabId (sometimes it is correct, sometimes it is 0).
+      // Thus, let the content_script fire it.
+      case 'onCommitted': {
+        const { domain, hostname } = parse(sender.url);
```

Comment:
This line has logic change (added `hostname`), others in the file are cosmetics.